.github/workflows/determine-workflow-runner.yml

```diff
@@ -0,0 +1,40 @@
+name: Determine Workflow Runner group
+
+on:
+  workflow_call:
+    inputs:
+      default_runner:
+        description: The runner type that is used by the calling workflow by default
+        required: true
+        type: string
+    outputs:
+      runner_group:
+        description: The runner all subsequent jobs within the calling workflow should run on
+        value: ${{ jobs.determine_workflow_runner.outputs.runner_group || inputs.default_runner }}
+
+env:
+  LARGE_RUNNER_GROUP_NAME: pl-4-core-large-runner
+
+jobs:
+  determine_workflow_runner:
+    runs-on: >-
+      ${{
+        (
+          github.event_name == 'pull_request'
+          && contains(github.event.pull_request.labels.*.name, 'urgent')
+        ) && 'pl-4-core-large-runner' || 'ubuntu-latest'
```

Comment:
I don't quite follow this sequence, what do the two values at the end do?

Reply:
It is an if statement:
```
if (github.event_name == 'pull_request' && contains(..) {
  runs-on: large runner
} else {
  runs-on: ubuntu-latest
}
```

Reply:
Oh I see, a bit obscure but it works 😅